components/greaselion/browser/greaselion_download_service.cc

```diff
@@ -100,11 +104,14 @@ GreaselionRule::~GreaselionRule() = default;
 
 void GreaselionRule::AddScriptAfterLoad(std::unique_ptr<std::string> contents,
                                         bool did_load) {
-  if (!did_load || !contents) {
+  if (did_load && contents)
+    scripts_.push_back(*contents);
+  else
     LOG(ERROR) << "Could not load Greaselion script";
-    return;
-  }
-  scripts_.push_back(*contents);
+  pending_scripts_ -= 1;
+  if (!pending_scripts_)
+    for (Observer& observer : observers_)
+      observer.OnRuleReady(this);
```

Comment:
if we're going to fire this even when we couldn't load the script https://github.com/brave/brave-core/pull/2748/files#diff-9aeb6794245eb42d1161fdac97c9a177R110 maybe we should also pass `did_load` through as a param?